dts/arm/nxp/nxp_imx.dtsi

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <dt-bindings/clock/imx_ccm.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	soc {
+
+		ddr_code: code@10000000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x10000000 0xfff0000>;
+			label = "DDR CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
```

Comment:
why does ddr_code have a clock-controller; and #clock-cells.  They should probalby be removed (same for all the various memory nodes).